client/src/components/testUtilities.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import { RenderResult, act, render } from '@testing-library/react'
+import { createLocation, createMemoryHistory, MemoryHistory } from 'history'
+import { match as routerMatch, Router } from 'react-router-dom'
+import equal from 'fast-deep-equal'
+import * as utilities from './utilities'
+
+type MatchParameter<Params> = { [K in keyof Params]?: string }
+
+const generateUrl = <Params extends MatchParameter<Params>>(
+  path: string,
+  params: Params
+): string => {
+  let tempPath = path
+
+  for (const param in params) {
+    /* istanbul ignore else */
+    if (Object.prototype.hasOwnProperty.call(params, param)) {
+      const value = params[param]
+      tempPath = tempPath.replace(`:${param}`, value as NonNullable<
+        typeof value
+      >)
+    }
+  }
+
+  return tempPath
+}
+
+export const routerTestProps = <Params extends MatchParameter<Params> = {}>(
+  path: string,
+  params: Params
+) => {
+  const match: routerMatch<Params> = {
+    isExact: false,
+    path,
+    url: generateUrl(path, params),
+    params,
+  }
+  const history = createMemoryHistory()
+  const location = createLocation(match.url)
+
+  return { history, location, match }
+}
+
+// Copied from https://testing-library.com/docs/example-react-router
+export const renderWithRouter = (
+  ui: React.ReactElement,
+  {
+    route = '/',
+    history = createMemoryHistory({ initialEntries: [route] }),
+  }: { route?: string; history?: MemoryHistory } = {}
+) => {
+  const Wrapper: React.FC = ({ children }: { children?: React.ReactNode }) => (
+    <Router history={history}>{children}</Router>
+  )
+
+  return {
+    ...render(ui, { wrapper: Wrapper }),
+    // Adding `history` to the returned utilities to allow us
+    // to reference it in our tests (just try to avoid using
+    // this to test implementation details).
+    history,
+  }
+}
```

Comment:
@MorganLove here's one new test helper I found in the docs. Seems like a good starting point for simplifying rendering components that rely on route params.